src/KernelApplication.cs

```diff
@@ -259,21 +306,43 @@ public int InstallKernelSpec(bool develop, LogLevel logLevel, string prefix = nu
 
             // Make a temporary directory to hold the kernel spec.
             var tempKernelSpecDir = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+            var filesToDelete = new List<string>();
             var jsonPath = Path.Combine(tempKernelSpecDir, "kernel.json");
             Directory.CreateDirectory(tempKernelSpecDir);
             File.WriteAllText(jsonPath, JsonConvert.SerializeObject(kernelSpec));
+            filesToDelete.Add(jsonPath);
             kernelSpecDir = tempKernelSpecDir;
 
+            // Add any additional files we may need.
+            if (additionalFiles != null) {
+                foreach (var (fileName, streamAction) in additionalFiles)
+                {
+                    var dest = Path.Combine(tempKernelSpecDir, fileName);
+                    var sourceStream = streamAction();
+                    using (var destStream = File.OpenWrite(dest))
+                    {
+                        sourceStream.CopyTo(destStream);
+                    }
+                    filesToDelete.Add(dest);
+                }
+            }
+
             // Find out if we need any extra arguments.
-            var extraArgs = prefix == null ? null : $"--prefix=\"{prefix}\"";
+            var extraArgs = new List<string>();
```

Comment:
we're still not providing kernels the ability to provide their own extraArgs. For example, iqsharp wants to specify it's cacheFolder name...

Reply:
In this case, the extra args are to the  `jupyter kernelspec install` command, but I agree, it would be reasonable to add extra arguments to the `kernelspec.json` file itself.

Reply:
This should now be implemented. I've modified the echo example to enable an install-time `--shout` option.